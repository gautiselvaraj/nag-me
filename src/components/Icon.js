import React from 'react';
import styled from 'styled-components';

const Icon = styled.i`
  ${props => props.inverse ? `color: ${props.theme.white}` : ''};
  font-family: icons;
  font-size: ${props => props.big ? '1.2rem' : '1rem'};
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  line-height: 1;
  speak: none;
  text-transform: none;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;

  &:before {
    content: ${props => props.edit ? '"\\e906"' :
              props.delete ? '"\\e9ac"' :
              props.question ? '"\\ea09"' :
              props.add ? '"\\ea0a"' :
              props.resume ? '"\\ea1c"' :
              props.pause ? '"\\ea1d"' :
              props.search ? '"\\e986"' : ''}
  }
`;

export default props => <Icon {...props}></Icon>;
