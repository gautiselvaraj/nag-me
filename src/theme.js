export default {
  main:           '#214154',
  mainDarken:     '#083148',
  mainLighten:    '#215a7b',
  black:          '#111111',
  white:          '#f6f6f6',
  greyLightest:   '#efefef',
  greyLighter:    '#dddddd',
  greyLight:      '#a5a5a5',
  grey:           '#898989',
  greyDark:       '#777777',
  greyDarker:     '#595959',
  greyDarkest:    '#222',
}
