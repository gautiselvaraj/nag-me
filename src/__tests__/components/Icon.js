import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Icon from '../../components/Icon';
import theme from '../../theme';
import 'jest-styled-components';

describe('<Icon />', () => {
  describe('default icon', () => {
    it('should match snapshot', () => {
      const tree = renderer.create(<Icon edit theme={theme} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should have correct element, content before, text & font-size', () => {
      let icon = mount(<Icon edit theme={theme} />);
      expect(icon.text()).toBe('');
      expect(icon.find('i').length).toBe(1);
      expect(icon).toHaveStyleRule('font-size', '1rem');
      icon = mount(<Icon edit big theme={theme} />);
      expect(icon).toHaveStyleRule('font-size', '1.2rem');
      expect(icon).toHaveStyleRule('content', '"\\e906"', {
        modifier: ':before'
      });
    });
  });

  describe('inverse link', () => {
    it('should match snapshot', () => {
      const tree = renderer
        .create(<Icon edit inverse theme={theme} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should have correct element & color', () => {
      const icon = mount(<Icon edit inverse theme={theme} />);
      expect(icon.find('i').length).toBe(1);
      expect(icon).toHaveStyleRule('color', theme.white);
    });
  });
});
