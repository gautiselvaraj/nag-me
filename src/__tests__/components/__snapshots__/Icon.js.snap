// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> default icon should match snapshot 1`] = `
.c0 {
  font-family: icons;
  font-size: 1rem;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  line-height: 1;
  speak: none;
  text-transform: none;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c0:before {
  content: "\\e906";
}

<i
  className="c0"
/>
`;

exports[`<Icon /> inverse link should match snapshot 1`] = `
.c0 {
  color: #f6f6f6;
  font-family: icons;
  font-size: 1rem;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  line-height: 1;
  speak: none;
  text-transform: none;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c0:before {
  content: "\\e906";
}

<i
  className="c0"
/>
`;
